library/Core.cpp

```diff
@@ -2419,11 +2419,22 @@ bool Core::SelectHotkey(int sym, int modifiers)
                                         binding.modifiers, modifiers);
                 continue;
             }
-            string focusString = Gui::getFocusString(screen);
-            if (!binding.focus.empty() && !prefix_matches(binding.focus, focusString)) {
-                DEBUG(keybinding).print("skipping keybinding due to focus string mismatch: '%s' !~ '%s'\n",
-                                        focusString.c_str(), binding.focus.c_str());
-                continue;
+            if (!binding.focus.empty()) {
+                // TODO: understand more about this to figure out if this solution works
+                bool found = false;
+                std::vector<std::string> focusStrings = Gui::getFocusString(Core::getTopViewscreen());
+                // is there convention for when to use size_t vs int?
+                for (size_t ii = 0; ii < focusStrings.size(); ii++) {
```

Comment:
```suggestion
                for (std::string focusString : focusStrings) {
```

Reply:
the standard is to use foreach syntax when possible, but if not then the var should be size_t and cast where necessary.